context.go

```diff
@@ -69,9 +71,10 @@ func NewContext(opt ...ContextOption) *Context {
 	ctxMutex.Unlock()
 
 	ctx := &Context{
-		ref: ref,
-		ptr: C.NewContext(opts.iso.ptr, opts.gTmpl.ptr, C.int(ref)),
-		iso: opts.iso,
+		EmbedderContext: context.Background(),
+		ref:             ref,
+		ptr:             C.NewContext(opts.iso.ptr, opts.gTmpl.ptr, C.int(ref)),
+		iso:             opts.iso,
```

Comment:
Is it possible to check the context option arguments to see if a context.Context is provided? If not provided, then the constructor can fallback to the background context.

Reply:
It doesn't really need to be set in the constructor.  We can just let them set the field after the context is constructed.